app/controllers/suggestion_controller.rb

```diff
@@ -125,66 +156,32 @@ def approve_suggestion
-          if @topic_id.nil?
-            #clean waitlists
-            SignedUpTeam.where(team_id: @team_id, is_waitlisted: 1).destroy_all
-            SignedUpTeam.create(topic_id: @signuptopic.id, team_id: @team_id, is_waitlisted: 0)
-          else
-            @signuptopic.private_to = @user_id
-            @signuptopic.save
-            #if this team has topic, Expertiza will send an email (suggested_topic_approved_message) to this team
-            proposer = User.find(@user_id)
-            teams_users = TeamsUser.where(team_id: @team_id)
-            cc_mail_list = Array.new
-            teams_users.each do |teams_user|
-              cc_mail_list << User.find(teams_user.user_id).email if teams_user.user_id != proposer.id
-            end
-            Mailer.suggested_topic_approved_message(
-            { to: proposer.email,
-              cc: cc_mail_list,
-              subject: "Suggested topic '#{@suggestion.title}' has already been approved",
-              body: {
-                approved_topic_name: @suggestion.title,
-                proposer: proposer.name
-              }
-            }
-            ).deliver
-          end
-        end
-      else
-        #if this team has topic, Expertiza will send an email (suggested_topic_approved_message) to this team
-        proposer = User.find(@user_id)
-        teams_users = TeamsUser.where(team_id: @team_id)
-        cc_mail_list = Array.new
-        teams_users.each do |teams_user|
-          cc_mail_list << User.find(teams_user.user_id).email if teams_user.user_id != proposer.id
+    if @suggestion.signup_preference == 'Y'
+      #if this user do not have team in this assignment, create one for him/her and assign this topic to this team.
+      if @team_id.nil?
+        create_new_team
+      else #this user has a team in this assignment, check whether this team has topic or not
+        if @topic_id.nil?
+          #clean waitlists
+          SignedUpTeam.where(team_id: @team_id, is_waitlisted: 1).destroy_all
+          SignedUpTeam.create(topic_id: @signuptopic.id, team_id: @team_id, is_waitlisted: 0)
+        else
+          @signuptopic.private_to = @user_id
+          @signuptopic.save
+          #if this team has topic, Expertiza will send an email (suggested_topic_approved_message) to this team
+          send_email
         end
-        Mailer.suggested_topic_approved_message(
-        { to: proposer.email,
-          cc: cc_mail_list,
-          subject: "Suggested topic '#{@suggestion.title}' has already been approved",
-          body: {
-            approved_topic_name: @suggestion.title,
-            proposer: proposer.name
-          }
-        }
-        ).deliver
       end
+    else
+      #if this team has topic, Expertiza will send an email (suggested_topic_approved_message) to this team
+      send_email
     end
-    redirect_to :action => 'show', :id => @suggestion
+  end
+
+  def approve_suggestion
```

Comment:
There is an approve and an approve_suggestion method, is this sort of confusing? At least one of them should be private.